File: fragments/solana_airdrop/solana_airdrop_utils.ts
import { Address, lamports } from "@solana/kit";
import { initRpcClient } from "../solana_rpc/solana_rpc_utils";
import { confirmRecentSignature } from "../solana_transaction/solana_transaction_utils";

export const sendAndConfirmAirdrop = async (address: Address, amount: bigint) => {
  const client = initRpcClient();
  const airdropAmount = lamports(amount);

  console.log(`Airdropping ${new Intl.NumberFormat().format(amount)} lamports to ${address}`);

  const sig = await client.requestAirdrop(address, airdropAmount, { commitment: "confirmed" }).send();
  const isConfirmed = await confirmRecentSignature(sig);

  if (!isConfirmed) {
    throw new Error(`Airdrop was not confirmed for ${address} with signature ${sig}`);
  }

  console.log(`Airdrop confirmed for ${address} with signature ${sig}`);
};

File: fragments/solana_airdrop/solana_airdrop_utils.py
from solders.pubkey import Pubkey
from fragments.solana_rpc import init_rpc_client
from fragments.solana_transaction import confirm_recent_signature


async def send_and_confirm_airdrop(address: Pubkey, amount: int) -> None:
    client = init_rpc_client()
    formatted_amount = f"{amount:,}"
    print(f"Airdropping {formatted_amount} lamports to {address}")

    sig = (await client.request_airdrop(address, amount)).value
    is_confirmed = await confirm_recent_signature(sig)

    if not is_confirmed:
        raise RuntimeError(f"Airdrop was not confirmed for {address} with signature {sig}")

    print(f"Airdrop confirmed for {address} with signature {sig}")
